primitives/consensus/babe/src/digests.rs

```diff
@@ -18,175 +18,101 @@
 
 #[cfg(feature = "std")]
 use super::{BABE_ENGINE_ID, AuthoritySignature};
-#[cfg(not(feature = "std"))]
-use super::{VRF_OUTPUT_LENGTH, VRF_PROOF_LENGTH};
 use super::{AuthorityId, AuthorityIndex, SlotNumber, BabeAuthorityWeight};
 #[cfg(feature = "std")]
 use sp_runtime::{DigestItem, generic::OpaqueDigestItemId};
 #[cfg(feature = "std")]
-use std::fmt::Debug;
+use std::{fmt::Debug, convert::{TryFrom, TryInto}};
 use codec::{Decode, Encode};
 #[cfg(feature = "std")]
-use codec::{Codec, Input, Error};
-#[cfg(feature = "std")]
-use schnorrkel::{
-	SignatureError, errors::MultiSignatureStage,
-	vrf::{VRFProof, VRFOutput, VRF_OUTPUT_LENGTH, VRF_PROOF_LENGTH}
-};
+use codec::Codec;
 use sp_std::vec::Vec;
-
+use sp_consensus_vrf::schnorrkel::{self, Randomness};
+#[cfg(feature = "std")]
+use sp_consensus_vrf::schnorrkel::SignatureError;
 
 /// A BABE pre-runtime digest. This contains all data required to validate a
 /// block and for the BABE runtime module. Slots can be assigned to a primary
 /// (VRF based) and to a secondary (slot number based).
-#[cfg(feature = "std")]
-#[derive(Clone, Debug)]
-pub enum PreDigest {
+#[derive(Clone, Debug, Encode, Decode)]
+pub enum RawPreDigest<VRFOutput=schnorrkel::RawVRFOutput, VRFProof=schnorrkel::RawVRFProof> {
 	/// A primary VRF-based slot assignment.
+	#[codec(index = "1")]
 	Primary {
-		/// VRF output
-		vrf_output: VRFOutput,
-		/// VRF proof
-		vrf_proof: VRFProof,
 		/// Authority index
 		authority_index: super::AuthorityIndex,
 		/// Slot number
 		slot_number: SlotNumber,
+		/// VRF output
+		vrf_output: VRFOutput,
```

Comment:
This breaks compatibility with existing chain such as Kusama, no?

Reply:
I see - as long as `vrf_output` and `vrf_proof` encode the same as before (i.e. they are thin wrappers) then it should all be OK.

Reply:
But isn't changing the order of fields problematic?

Reply:
I had the same initial thought, but this is actually the same order of fields as the `RawBabePreDigest` which is removed in this PR.